src/app/calculator/pumps/achievable-efficiency/achievable-efficiency.component.ts

```diff
@@ -71,44 +73,82 @@ export class AchievableEfficiencyComponent implements OnInit {
     if (!this.psat) {
       this.achievableEfficiencyService.flowRate = this.efficiencyForm.controls.flowRate.value;
       this.achievableEfficiencyService.pumpType = this.efficiencyForm.controls.pumpType.value;
+      this.pumpEfficiencyInputs = this.achievableEfficiencyService.getObjectFromForm(this.efficiencyForm);
+      this.achievableEfficiencyService.pumpEfficiencyInputs = this.pumpEfficiencyInputs;
```

Comment:
1. I can't see a reason to assign a class property value on destroy. 
2. The pattern to assign service.inputs from the form is also a little weird but I assume this is an old MEASUR pattern as head tool does it. Errors coming from the in-assessment calculator could be related to this.